src/features/embalm/stepContent/steps/CreateSarcophagus.tsx

```diff
@@ -42,11 +42,13 @@ export function CreateSarcophagus() {
             currentStage={currentStage}
             stageError={stageError}
           >
-            <ProgressTrackerStage>Connect to Archaeologists</ProgressTrackerStage>
-            <ProgressTrackerStage>Upload Archaeologist Data to Arweave</ProgressTrackerStage>
-            <ProgressTrackerStage>Retrieve Archaeologist Signatures</ProgressTrackerStage>
-            <ProgressTrackerStage>Upload File Data to Arweave</ProgressTrackerStage>
-            <ProgressTrackerStage>Create Sarcophagus</ProgressTrackerStage>
+            {Object.values(createSarcophagusStages)
+              // Necessarily, a couple of these mappings don't have UI importance, thus no titles.
+              // We exclude those, and, if already approved, also exclude the approve stage
+              .filter(text => !!text && (hasApproved ? text !== createSarcophagusStages[CreateSarcophagusStage.APPROVE] : true))
+              .map(stage =>
+                <ProgressTrackerStage key={stage}>{stage}</ProgressTrackerStage>
+              )}
```

Comment:
Just wanted to share the original intent here and how things may have changed. When I wrote this I was trying to avoid a mapping here, so I manually added the `stageStatus` and `index` props to the children in the `ProgressTracker` component like so.
```
  const childrenWithProps = React.Children.map(children, (child, index) => {
    if (React.isValidElement(child)) {
      return React.cloneElement<ProgressTrackerStageProps>(child, {
        // index is a number but is being used as a value of the CreateSarcophagusStage enum
        stageStatus: getStageStatus(index + 1),
        index,
        stageError,
      });
    }
```
This made the `ProgressTracker` component a bit more complicated in exchange for making its implementation simpler, by not having to programmatically map over an object and allowing the static titles to exist within the return statement of the component. However, that didn't accommodate for a dynamic list of stages, which is what the mapping is for.

There are other solutions, but if we are going with a mapping here the `ProgressTracker` should be modified to remove the unnecessary complication.